File: 11기 코딩테스트_230308/연습1/Fibonacci/Fibonacci_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.supereasy.Fibonacci.Fibonacci import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        n = 6

        # when
        result = solution(n)

        # then
        self.assertEqual(8, result)

    def test_2(self):
        # given
        n = 1

        # when
        result = solution(n)

        # then
        self.assertEqual(1, result)

    def test_3(self):
        # given
        n = 2

        # when
        result = solution(n)

        # then
        self.assertEqual(1, result)

    def test_4(self):
        # given
        n = 10

        # when
        result = solution(n)

        # then
        self.assertEqual(55, result)

    def test_5(self):
        # given
        n = 33

        # when
        result = solution(n)

        # then
        self.assertEqual(3524578, result)

    def test_6(self):
        # given
        n = 40

        # when
        result = solution(n)

        # then
        self.assertEqual(102334155, result)


if __name__ == '__main__':
    unittest.main()

File: 11기 코딩테스트_230308/연습1/Fibonacci/Fibonacci_test.js
import assert from 'assert'
import solution from './Fibonacci.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const n = 6

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual(8, result)
    done();
  });

  it('test 2', function (done) {
    // given
    const n = 1

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual(1, result)
    done();
  });

  it('test 3', function (done) {
    // given
    const n = 2

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual(1, result)
    done();
  });

  it('test 4', function (done) {
    // given
    const n = 10

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual(55, result)
    done();
  });

  it('test 5', function (done) {
    // given
    const n = 33

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual(3524578, result)
    done();
  });

  it('test 6', function (done) {
    // given
    const n = 40

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual(102334155, result)
    done();
  });
})
